doc/source/whatsnew/v2.1.0.rst

```diff
@@ -111,6 +111,7 @@ Other enhancements
 - Improve error message when setting :class:`DataFrame` with wrong number of columns through :meth:`DataFrame.isetitem` (:issue:`51701`)
 - Improved error handling when using :meth:`DataFrame.to_json` with incompatible ``index`` and ``orient`` arguments (:issue:`52143`)
 - Improved error message when creating a DataFrame with empty data (0 rows), no index and an incorrect number of columns. (:issue:`52084`)
+- Improved error message when creating a Timedelta from a ambiguous time period (:issue:`54059`)
```

Comment:
We don't need a whatsnew entry for this change since the original change was going to be new in 2.1